llm/memory.go

```diff
@@ -305,6 +307,30 @@ func EstimateGPULayers(gpus []gpu.GpuInfo, ggml *GGML, projectors []string, opts
 	return estimate
 }
 
+// estimateKvCacheSize determines the memory required for K or V cache based on the quantization type
+func estimateKvCacheSize(cacheType string, numCtx, blockCount, embeddingHeadCount, headCountKV uint64) uint64 {
```

Comment:
Thanks for updating the memory estimation. Two comments:

1. Would it be possible to add some unit tests for this?
2. Wouldn't it make sense for this function to stick to how many bytes are required (vs bringing in the final estimate as well)

Reply:
Yep, no worries, good idea!

Reply:
Added tests https://github.com/ollama/ollama/pull/6279/files#diff-bd3b44da775c8156c7acd436c9a446c0e22103c6890f76e86cd27b5f198146abR131